v2/apiserver/internal/core/logs.go

```diff
@@ -168,6 +169,20 @@ func (l *logsService) Stream(
 				}
 			}
 		}
+
+		// Check to see if we need to look up logs via a specific event ID,
+		// as job may be cached and carried over on a retry event
+		if job.Status != nil && job.Status.LogsEventID != "" {
+			event, err = l.eventsStore.Get(ctx, job.Status.LogsEventID)
+			if err != nil {
+				return nil,
+					errors.Wrapf(
+						err,
+						"error retrieving original event %q from store",
+						job.Status.LogsEventID,
+					)
```

Comment:
If the event is gone, the error we get back with be a `*meta.ErrNotFound`. Because that's the innermost of the nested errors, the client will get a 404 with a message body that indicates <event id of original event> not found. That could end up being pretty misleading...

I am wondering if we should trap `*meta.ErrNotFound`, specifically, here and raise an all new error that will be more informative.

Reply:
This appears to still be an issue.

Reply:
Updated here -- did I address the concern as envisioned?